dali/benchmark/one_hot_bench.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2021, Maksymilian Grochowski. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <benchmark/benchmark.h>
+#include "dali/benchmark/operator_bench.h"
+#include "dali/benchmark/dali_bench.h"
+
+namespace dali {
+
+static void OneHotGPUArgs(benchmark::internal::Benchmark *b) {
+  for (int batch_size = 256; batch_size >= 1; batch_size /= 2) {
+    for (int H = 1000; H >= 500; H /= 2) {
```

Comment:
This benchmark does not reflect real-world usage. Typically, there are few input values (definitely not in the 1000x1000 range) but typically there are much more classes (1000 for ImageNet).